File: 0136_single_number/solution.c
#include <stdlib.h>

// runtime: 100.00%, memory: 32.09%
int singleNumber(int* nums, int numsSize) {
    int single = 0;
    for (int i = 0; i < numsSize; i++) {
        single ^= nums[i];
    }

    return single;
}

File: 0136_single_number/Solution.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, Boolean> map = new HashMap<>();
        int single = -1;
        for (int i = 0;  i < nums.length; i++) { //int j = nums.length; i < nums.length, j >= 0; i++; j--
            if (map.getOrDefault(nums[i], null) == null)
                map.put(nums[i], false);
            else
                map.put(nums[i], true);
        }
        for (int i = 0; i < nums.length; i++) {
            if (map.get(nums[i]) == false)
                single = nums[i];
        }
        return single;
    }
}